extensions/objects/models/objects.py

```diff
@@ -1633,208 +1552,265 @@ def get_schema(cls):
+    """List of sets of TranslatableHtml content ids."""
+
+    default_value = []
+
+    @classmethod
+    def get_schema(cls):
+        """Returns the object schema.
+
+        Returns:
+            dict. The object schema.
+        """
+        return {
+            'type': 'list',
+            'items': SetOfTranslatableHtmlContentIds.get_schema()
+        }
+
+
+class BaseTranslatableObject(BaseObject):
+    """Base translatable object class.
+
+    This is a superclass for objects that are translatable and thus require a
+    content id. This class enforces that the object is a dictionary with a
+    content id field. The schema of the actual value is determined by the
+    _value_schema property.
+    """
+
+    # The key name in the translatable object corresponding to the translatable
+    # value. This field must be populated by subclasses.
+    _value_key_name = None
+    # The schema of the translatable value. This field must be populated by
+    # subclasses.
+    _value_schema = None
+    # The default value of the object. This field must be populated by
+    # subclasses.
+    default_value = None
+
+    @staticmethod
+    def normalize_value(value):
+        """Normalizes the translatable value of the object.
+
+        Args:
+            value: *. The translatable part of the Python object (corresponding
+                to the non-content-id field) which is to be normalized.
+
+        Returns:
+            *. The normalized value.
+        """
+        raise NotImplementedError(
+            'Subclasses of BaseTranslatableObject should implement '
+            'normalize_value().')
+
+    @classmethod
+    def get_schema(cls):
+        """Returns the full object schema.
+
+        Returns:
+            dict. The object schema.
+        """
+        if cls._value_key_name is None or cls._value_schema is None:
+            raise NotImplementedError(
+                'The _value_key_name and _value_schema for this class must '
+                'both be set.')
+        return {
+            'type': 'dict',
+            'properties': [{
+                'name': 'contentId',
+                # The default content id is none. However, it should be
+                # populated before being saved. The normalize() method has
+                # validation checks for this.
+                'schema': {'type': 'unicode_or_none'}
+            }, {
+                'name': cls._value_key_name,
+                'schema': copy.deepcopy(cls._value_schema),
+            }]
+        }
+
+    @classmethod
+    def normalize(cls, raw):
+        """Validates and normalizes a raw Python object.
+
+        Args:
+            raw: dict. A Python object to be validated against the schema,
+                normalizing if necessary.
+
+        Returns:
+            dict. The normalized object.
+
+        Raises:
+            TypeError. Error while normalizing.
+        """
+        if not isinstance(raw['contentId'], python_utils.BASESTRING):
```

Comment:
should we call the normalize of superclass I think that will handle all the validation using `get_schema`?

Reply:
That's a great point, I completely missed that! Simplified.